### PR TITLE
8 refactor for package

### LINKED ISSUE #8: add `setup.cfg` and test pip install editable mode
**Description of the Feature to be Added**

Add a `setup.cfg` file to allow developers to install contents of `src` in editable mode.

**(OPTIONAL) Suggested Implementations**

**Additional context**

### PR DESCRIPTION
**Pull request by ethan-moss**
## Description

Refactored repo for use as a package, and added a short set-up test to check this is done sucessfully:

- `.pre-commit-config.yaml` updated pydocstyle to ignore D203 errors which is deprecated
- `CONTRIBUTING.md` updated to show installation in editable mode, and pytest --runsetup.
- `conftest.py` added to handle switching of setup tests
- `pyproject.toml` updated to install src folder in editable mode
- `requirements.txt` updated with pytest
- `tests/test_setup.pt` added to check installation. marked to run only when --runsetup is added to pytest cli call
- `src/heimdall_transport` added and refactored content in this dir for installation in editable mode

Fixes #8 

## Motivation and Context

Refactored the repo to all package installation in editable mode to made development easier. Also added a simple setup test, which pytest cli flag switching such that the setup tests are marked to only run when the `--runsetup` flag is added.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Pytest run (with `--runsetup` flag) and passes test as expected. Without flag, the test does not run - expected behaviour. `src/heimdall_transport/utils/check_setup.py` still runs as expected (see #7 and Additional Notes). 

Test configuration details:
* OS: macOS 
* Python version: 3.9.13
* Java version: AdoptOpenJDK-11
* Python management system: venv = conda, packages = pip

## Advice for reviewer

Should now be able to install as a package in editable using `pip install -e .`. `CONTRIBUTING.md` updated to reflect the steps. Expected behaviour: Before installation, running `pytest --runsetup` should fail. After installation, is should fail work.

## Checklist:

- [X] My code follows the intended structure of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

## Additional comments

#7 can be added to `test_setup.py` to benefit from the `--runsetup` pytest cli flag.

### CODE REVIEW

**Comment by r-leyshon** [comment]
Morning chaps. Is this being handled? I can see it's in the 'in review' list on the kanban, but nobody is assigned? Happy to pick up but just paused incase Serg was handling it.
@ethan-moss  @SergioRec

**Comment by ethan-moss** [comment]
Morning @r-leyshon, @SergioRec! Yes, this one is ready for review but no-one is assigned just yet. Feel free to take a look. Thanks for your help and if I can clarify anything just let me know.

File: tests/test_setup.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        This is a simple test to check the package has been installed and is
+        available for use.
+        """
+        try:
+            import heimdall_transport as ht
+
+            assert ht.__name__ == "heimdall_transport"
+        except ImportError:
+            pytest.fail("Unable to find `heimdall_transport`.")
```
Comment (r-leyshon) [inline tests/test_setup.py:31]: Failing for me:

```
>>> import heimdall_transport as ht
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'heimdall_transport'
>>> import src.heimdall_transport as ht
>>> ht.__name__
'src.heimdall_transport'
```
I didn't expect to see a src folder any more. Suggest ditching it in favour of `heimdall_transport`.

**Review by r-leyshon — DISMISSED** [review]
@ethan-moss  Please see comment in regards to src folder suggested change: `src` -> `heimdall_transport`

File: tests/test_setup.py (hunk 1 above)
Comment (ethan-moss) [reply tests/test_setup.py:31]: The src -> package folder structure is a standard pypi layout: https://packaging.python.org/en/latest/tutorials/packaging-projects/

Have you called `pip install -e .` before running pytest? I would expect pytest to fail before this is done, and then work afterwards. It's a check to make sure `heimdall_transport` has been installed and is available. I checked pytest with and without it installed and pytest performed as expected.

Comment (r-leyshon) [reply tests/test_setup.py:31]: added editable install req to deps

**Review by r-leyshon — APPROVED** [review]

**Pull request merged**